<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="2" total="4" passed="2">
  <reporter-output>
  </reporter-output>
  <suite name="Command line suite" duration-ms="463" started-at="2017-03-20T08:57:47Z" finished-at="2017-03-20T08:57:47Z">
    <groups>
    </groups>
    <test name="Command line test" duration-ms="463" started-at="2017-03-20T08:57:47Z" finished-at="2017-03-20T08:57:47Z">
      <class name="ass.kasimyur.cvi3.functional.AverageStockPriceTest">
        <test-method status="PASS" signature="average()[pri:0, instance:ass.kasimyur.cvi3.functional.AverageStockPriceTest@19bb089b]" name="average" duration-ms="380" started-at="2017-03-20T09:57:47Z" finished-at="2017-03-20T09:57:47Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- average -->
      </class> <!-- ass.kasimyur.cvi3.functional.AverageStockPriceTest -->
      <class name="ass.kasimyur.cvi3.stock.trader.TraderTest">
        <test-method status="PASS" signature="setup()[pri:0, instance:ass.kasimyur.cvi3.stock.trader.TraderTest@5a39699c]" name="setup" is-config="true" duration-ms="3" started-at="2017-03-20T09:57:47Z" finished-at="2017-03-20T09:57:47Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="PASS" signature="testShouldFindCorrectMinimum()[pri:0, instance:ass.kasimyur.cvi3.stock.trader.TraderTest@5a39699c]" name="testShouldFindCorrectMinimum" duration-ms="53" started-at="2017-03-20T09:57:47Z" finished-at="2017-03-20T09:57:47Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testShouldFindCorrectMinimum -->
        <test-method status="FAIL" signature="testShouldFindCorrectMinimumFast()[pri:0, instance:ass.kasimyur.cvi3.stock.trader.TraderTest@5a39699c]" name="testShouldFindCorrectMinimumFast" duration-ms="5" started-at="2017-03-20T09:57:47Z" finished-at="2017-03-20T09:57:47Z">
          <exception class="java.util.concurrent.RejectedExecutionException">
            <message>
              <![CDATA[Task java.util.concurrent.ExecutorCompletionService$QueueingFuture@19328f63 rejected from java.util.concurrent.ThreadPoolExecutor@17f56235[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ExecutorCompletionService$QueueingFuture@19328f63 rejected from java.util.concurrent.ThreadPoolExecutor@17f56235[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1369)
	at java.util.concurrent.ExecutorCompletionService.submit(ExecutorCompletionService.java:181)
	at ass.kasimyur.cvi3.stock.trader.solution.ExecutorTrader.suggestStockToBuy(ExecutorTrader.java:52)
	at ass.kasimyur.cvi3.stock.trader.TraderTest.testShouldFindCorrectMinimumFast(TraderTest.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:46)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:37)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
]]>
            </full-stacktrace>
          </exception> <!-- java.util.concurrent.RejectedExecutionException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testShouldFindCorrectMinimumFast -->
        <test-method status="FAIL" signature="testShouldSkipFailed()[pri:0, instance:ass.kasimyur.cvi3.stock.trader.TraderTest@5a39699c]" name="testShouldSkipFailed" duration-ms="1" started-at="2017-03-20T09:57:47Z" finished-at="2017-03-20T09:57:47Z">
          <exception class="java.util.concurrent.RejectedExecutionException">
            <message>
              <![CDATA[Task java.util.concurrent.ExecutorCompletionService$QueueingFuture@312a12c2 rejected from java.util.concurrent.ThreadPoolExecutor@17f56235[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ExecutorCompletionService$QueueingFuture@312a12c2 rejected from java.util.concurrent.ThreadPoolExecutor@17f56235[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1369)
	at java.util.concurrent.ExecutorCompletionService.submit(ExecutorCompletionService.java:181)
	at ass.kasimyur.cvi3.stock.trader.solution.ExecutorTrader.suggestStockToBuy(ExecutorTrader.java:52)
	at ass.kasimyur.cvi3.stock.trader.TraderTest.testShouldSkipFailed(TraderTest.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:46)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:37)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
]]>
            </full-stacktrace>
          </exception> <!-- java.util.concurrent.RejectedExecutionException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testShouldSkipFailed -->
      </class> <!-- ass.kasimyur.cvi3.stock.trader.TraderTest -->
    </test> <!-- Command line test -->
  </suite> <!-- Command line suite -->
</testng-results>
